File: hadoop + Hive + Kafka + frontend/producer.py
# producer.py
from kafka import KafkaProducer
import json
import time
import random

# Initialize Kafka producer
producer = KafkaProducer(
    bootstrap_servers='localhost:9092',
    value_serializer=lambda v: json.dumps(v).encode('utf-8')
)

# Define the topic
topic = 'air_quality_data'

# Generate and send data
while True:
    data = {
        'timestamp': time.time(),
        'aqi': random.randint(0, 500)  # Simulate AQI values
    }
    producer.send(topic, data)
    print(f"Sent: {data}")
    time.sleep(1)  # Send data every second

File: dynamic charts/producer.js
const { Kafka } = require('kafkajs');

const kafka = new Kafka({
  clientId: 'chart-producer',
  brokers: ['localhost:9092']
});

const producer = kafka.producer();

const run = async () => {
  await producer.connect();
  let x = 1;

  setInterval(async () => {
    const y = Math.floor(Math.random() * 100);
    const message = { x, y };
    await producer.send({
      topic: 'chart-data',
      messages: [{ value: JSON.stringify(message) }],
    });
    console.log('Message sent:', message);
    x++;
  }, 1000);
};

run().catch(console.error);
